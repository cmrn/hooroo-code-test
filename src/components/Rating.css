.Rating {
  line-height: 30px;
}