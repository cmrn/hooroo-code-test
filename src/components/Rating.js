import React from 'react';
import PropTypes from 'prop-types';
import ReactStars from 'react-stars';
import cnames from 'classnames';
import './Rating.css';

const Rating = ({ rating, type, className, ...props }) => {
  const isStar = type === 'star'
  const char = isStar ? '★' : '●';
  const size = isStar ? 25 : 30; // Circle is smaller than star, so we need to scale it up
  const label = `${rating} stars${isStar ? '' : ' (self rated)'}`; // for screen readers
  return (
    <div aria-label={label} className={cnames(className, "Rating")} {...props}>
      <ReactStars char={char} count={5} value={rating} size={size} edit={false} color1={'#ccc'} />
    </div>
  );
}

Rating.propTypes = {
  rating: PropTypes.number.isRequired,
  type: PropTypes.oneOf(['star', 'self']).isRequired,
  className: PropTypes.string,
}

export default Rating;