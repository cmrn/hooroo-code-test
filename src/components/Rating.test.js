import React from 'react';
import renderer from 'react-test-renderer';
import Rating from './Rating';

beforeEach(() => {
  Math.random = jest.fn(() => 0.0487723387081509); // react-star uses a random number
})

it('renders 5 stars', () => {
  const tree = renderer.create(
    <Rating rating={5} type={'star'} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders half stars', () => {
  const tree = renderer.create(
    <Rating rating={3.5} type={'star'} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders 4.5 circles', () => {
  const tree = renderer.create(
    <Rating rating={4.5} type={'self'} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('allows a className to be set', () => {
  const tree = renderer.create(
    <Rating rating={5} type={'star'} className="SomeClass" />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('passes through other props', () => {
  const tree = renderer.create(
    <Rating rating={5} type={'star'} aria-label="another prop" />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
