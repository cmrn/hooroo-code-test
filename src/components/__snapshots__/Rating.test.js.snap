// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`allows a className to be set 1`] = `
<div
  aria-label="5 stars"
  className="SomeClass Rating"
>
  <div
    className={undefined}
    style={
      Object {
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
            .react-stars-00487723387081509:before {
              position: absolute;
              overflow: hidden;
              display: block;
              z-index: 1;
              top: 0; left: 0;
              width: 50%;
              content: attr(data-forhalf);
              color: #ffd700;
          }",
        }
      }
    />
    <span
      className=""
      data-forhalf="★"
      data-index={0}
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "color": "#ffd700",
          "cursor": "default",
          "display": "block",
          "float": "left",
          "fontSize": "25px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      ★
    </span>
    <span
      className=""
      data-forhalf="★"
      data-index={1}
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "color": "#ffd700",
          "cursor": "default",
          "display": "block",
          "float": "left",
          "fontSize": "25px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      ★
    </span>
    <span
      className=""
      data-forhalf="★"
      data-index={2}
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "color": "#ffd700",
          "cursor": "default",
          "display": "block",
          "float": "left",
          "fontSize": "25px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      ★
    </span>
    <span
      className=""
      data-forhalf="★"
      data-index={3}
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "color": "#ffd700",
          "cursor": "default",
          "display": "block",
          "float": "left",
          "fontSize": "25px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      ★
    </span>
    <span
      className=""
      data-forhalf="★"
      data-index={4}
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "color": "#ffd700",
          "cursor": "default",
          "display": "block",
          "float": "left",
          "fontSize": "25px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      ★
    </span>
  </div>
</div>
`;

exports[`passes through other props 1`] = `
<div
  aria-label="another prop"
  className="Rating"
>
  <div
    className={undefined}
    style={
      Object {
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
            .react-stars-00487723387081509:before {
              position: absolute;
              overflow: hidden;
              display: block;
              z-index: 1;
              top: 0; left: 0;
              width: 50%;
              content: attr(data-forhalf);
              color: #ffd700;
          }",
        }
      }
    />
    <span
      className=""
      data-forhalf="★"
      data-index={0}
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "color": "#ffd700",
          "cursor": "default",
          "display": "block",
          "float": "left",
          "fontSize": "25px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      ★
    </span>
    <span
      className=""
      data-forhalf="★"
      data-index={1}
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "color": "#ffd700",
          "cursor": "default",
          "display": "block",
          "float": "left",
          "fontSize": "25px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      ★
    </span>
    <span
      className=""
      data-forhalf="★"
      data-index={2}
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "color": "#ffd700",
          "cursor": "default",
          "display": "block",
          "float": "left",
          "fontSize": "25px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      ★
    </span>
    <span
      className=""
      data-forhalf="★"
      data-index={3}
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "color": "#ffd700",
          "cursor": "default",
          "display": "block",
          "float": "left",
          "fontSize": "25px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      ★
    </span>
    <span
      className=""
      data-forhalf="★"
      data-index={4}
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "color": "#ffd700",
          "cursor": "default",
          "display": "block",
          "float": "left",
          "fontSize": "25px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      ★
    </span>
  </div>
</div>
`;

exports[`renders 4.5 circles 1`] = `
<div
  aria-label="4.5 stars (self rated)"
  className="Rating"
>
  <div
    className={undefined}
    style={
      Object {
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
            .react-stars-00487723387081509:before {
              position: absolute;
              overflow: hidden;
              display: block;
              z-index: 1;
              top: 0; left: 0;
              width: 50%;
              content: attr(data-forhalf);
              color: #ffd700;
          }",
        }
      }
    />
    <span
      className=""
      data-forhalf="●"
      data-index={0}
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "color": "#ffd700",
          "cursor": "default",
          "display": "block",
          "float": "left",
          "fontSize": "30px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      ●
    </span>
    <span
      className=""
      data-forhalf="●"
      data-index={1}
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "color": "#ffd700",
          "cursor": "default",
          "display": "block",
          "float": "left",
          "fontSize": "30px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      ●
    </span>
    <span
      className=""
      data-forhalf="●"
      data-index={2}
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "color": "#ffd700",
          "cursor": "default",
          "display": "block",
          "float": "left",
          "fontSize": "30px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      ●
    </span>
    <span
      className=""
      data-forhalf="●"
      data-index={3}
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "color": "#ffd700",
          "cursor": "default",
          "display": "block",
          "float": "left",
          "fontSize": "30px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      ●
    </span>
    <span
      className="react-stars-00487723387081509"
      data-forhalf="●"
      data-index={4}
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "color": "#ccc",
          "cursor": "default",
          "display": "block",
          "float": "left",
          "fontSize": "30px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      ●
    </span>
  </div>
</div>
`;

exports[`renders 5 stars 1`] = `
<div
  aria-label="5 stars"
  className="Rating"
>
  <div
    className={undefined}
    style={
      Object {
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
            .react-stars-00487723387081509:before {
              position: absolute;
              overflow: hidden;
              display: block;
              z-index: 1;
              top: 0; left: 0;
              width: 50%;
              content: attr(data-forhalf);
              color: #ffd700;
          }",
        }
      }
    />
    <span
      className=""
      data-forhalf="★"
      data-index={0}
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "color": "#ffd700",
          "cursor": "default",
          "display": "block",
          "float": "left",
          "fontSize": "25px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      ★
    </span>
    <span
      className=""
      data-forhalf="★"
      data-index={1}
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "color": "#ffd700",
          "cursor": "default",
          "display": "block",
          "float": "left",
          "fontSize": "25px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      ★
    </span>
    <span
      className=""
      data-forhalf="★"
      data-index={2}
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "color": "#ffd700",
          "cursor": "default",
          "display": "block",
          "float": "left",
          "fontSize": "25px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      ★
    </span>
    <span
      className=""
      data-forhalf="★"
      data-index={3}
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "color": "#ffd700",
          "cursor": "default",
          "display": "block",
          "float": "left",
          "fontSize": "25px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      ★
    </span>
    <span
      className=""
      data-forhalf="★"
      data-index={4}
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "color": "#ffd700",
          "cursor": "default",
          "display": "block",
          "float": "left",
          "fontSize": "25px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      ★
    </span>
  </div>
</div>
`;

exports[`renders half stars 1`] = `
<div
  aria-label="3.5 stars"
  className="Rating"
>
  <div
    className={undefined}
    style={
      Object {
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
            .react-stars-00487723387081509:before {
              position: absolute;
              overflow: hidden;
              display: block;
              z-index: 1;
              top: 0; left: 0;
              width: 50%;
              content: attr(data-forhalf);
              color: #ffd700;
          }",
        }
      }
    />
    <span
      className=""
      data-forhalf="★"
      data-index={0}
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "color": "#ffd700",
          "cursor": "default",
          "display": "block",
          "float": "left",
          "fontSize": "25px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      ★
    </span>
    <span
      className=""
      data-forhalf="★"
      data-index={1}
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "color": "#ffd700",
          "cursor": "default",
          "display": "block",
          "float": "left",
          "fontSize": "25px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      ★
    </span>
    <span
      className=""
      data-forhalf="★"
      data-index={2}
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "color": "#ffd700",
          "cursor": "default",
          "display": "block",
          "float": "left",
          "fontSize": "25px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      ★
    </span>
    <span
      className="react-stars-00487723387081509"
      data-forhalf="★"
      data-index={3}
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "color": "#ccc",
          "cursor": "default",
          "display": "block",
          "float": "left",
          "fontSize": "25px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      ★
    </span>
    <span
      className=""
      data-forhalf="★"
      data-index={4}
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "color": "#ccc",
          "cursor": "default",
          "display": "block",
          "float": "left",
          "fontSize": "25px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      ★
    </span>
  </div>
</div>
`;
